File: plugin/testData/shared/basics/Interface.shared.kt
package com.test

interface MyInterface {
    val x: String get() = ""
    fun y(string: String): String {
        return this.x + string
    }
}

class ImplOverX: MyInterface {
    override val x: String
        get() = "Hello!"
}

class ImplOverY: MyInterface {
    override fun y(string: String): String {
        return "${this.x}!"
    }
}

class ImplBoth: MyInterface {
    override val x: String
        get() = "Hello!"
    override fun y(string: String): String {
        return "${this.x}!"
    }
}

fun main(){
    val items: List<MyInterface> = listOf(
        ImplBoth(),
        ImplOverX(),
        ImplOverY()
    )
    for(item in items){
        println(item.x)
        println(item.y("Input"))
    }
}

File: plugin/testDataSwift/shared/basics/Interface.shared.swift
//Package: com.test
//Converted using Khrysalis2

import Foundation



public protocol MyInterface {
    
    var x: String { get }
    
    func y(string: String) -> String
    func y(_ string: String) -> String
}

public extension MyInterface {
    public var x: String {
        get {
            return ""
        }
    }
    
    func y(string: String) -> String {
        return x + string
    }
    func y(_ string: String) -> String {
        return y(string: string)
    }
}
 
 

public class ImplOverX: MyInterface {
    
    
    public var x: String {
        get {
            return "Hello!"
        }
    }
    
    public init() {
    }
}
 
 

public class ImplOverY: MyInterface {
    
    
    
    public func y(string: String) -> String {
        return "\(x)!"
    }
    public func y(_ string: String) -> String {
        return y(string: string)
    }
    
    public init() {
    }
}
 
 

public class ImplBoth: MyInterface {
    
    
    public var x: String {
        get {
            return "Hello!"
        }
    }
    
    public func y(string: String) -> String {
        return "\(x)!"
    }
    public func y(_ string: String) -> String {
        return y(string: string)
    }
    
    public init() {
    }
}
 
 

public func main() -> Void {
    var items: Array<MyInterface> = [ImplBoth(), ImplOverX(), ImplOverY()]
    
    for item in items {
        print(item.x)
        print(item.y("Input"))
    }
}
 
